Index: build.py
===================================================================
--- build.py	(revision 1652)
+++ build.py	(working copy)
@@ -149,9 +149,10 @@
   def run(self):
     self.gen_core()
     self.gen_blocks()
-    self.gen_generator('javascript')
-    self.gen_generator('python')
-    self.gen_generator('dart')
+    #self.gen_generator('javascript')
+    #self.gen_generator('python')
+    #self.gen_generator('dart')
+    self.gen_generator('lua')
 
   def gen_core(self):
     target_filename = 'blockly_compressed.js'
